xrpl/core/binarycodec/types/xchain_bridge.py

```diff
@@ -0,0 +1,98 @@
+"""Codec for serializing and deserializing bridge fields."""
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, Tuple, Type, Union
+
+from xrpl.core.binarycodec.binary_wrappers.binary_parser import BinaryParser
+from xrpl.core.binarycodec.exceptions import XRPLBinaryCodecException
+from xrpl.core.binarycodec.types.account_id import AccountID
+from xrpl.core.binarycodec.types.issue import Issue
+from xrpl.core.binarycodec.types.serialized_type import SerializedType
+
+_TYPE_ORDER: List[Tuple[str, Type[SerializedType]]] = [
+    ("LockingChainDoor", AccountID),
+    ("LockingChainIssue", Issue),
+    ("IssuingChainDoor", AccountID),
+    ("IssuingChainIssue", Issue),
+]
+
+_TYPE_KEYS = {type[0] for type in _TYPE_ORDER}
+
+
+class XChainBridge(SerializedType):
+    """Codec for serializing and deserializing bridge fields."""
+
+    def __init__(self: XChainBridge, buffer: bytes) -> None:
+        """Construct a XChainBridge from given bytes."""
+        super().__init__(buffer)
+
+    @classmethod
+    def from_value(
+        cls: Type[XChainBridge], value: Union[str, Dict[str, str]]
+    ) -> XChainBridge:
+        """
+        Construct a XChainBridge object from a dictionary representation of a bridge.
+
+        Args:
+            value: The dictionary to construct a XChainBridge object from.
+
+        Returns:
+            A XChainBridge object constructed from value.
+
+        Raises:
+            XRPLBinaryCodecException: If the XChainBridge representation is invalid.
+        """
+        if isinstance(value, dict) and set(value.keys()) == _TYPE_KEYS:
+            buffer = b""
+            for name, object_type in _TYPE_ORDER:
+                obj = object_type.from_value(value[name])
+                if object_type == AccountID:
+                    buffer += bytes.fromhex("14")  # AccountID length (I think)
```

Comment:
Can you verify if 14 is correct then remove the "I think"?
```suggestion
                    buffer += bytes.fromhex("14")  # AccountID length
```